server/src/services/group_permissions.service.ts

```diff
@@ -139,6 +140,35 @@ export class GroupPermissionsService {
     } = body;
 
     await getManager().transaction(async (manager) => {
+      //update user group name
+      if (name) {
+        if (!name.trim()) {
+          throw new BadRequestException('Group name should not be empty');
+        }
+
+        const reservedGroups = ['admin', 'all_users'];
+        if (reservedGroups.includes(groupPermission.group)) {
+          throw new BadRequestException('Cannot update a default group name');
+        }
+
+        if (reservedGroups.includes(name.replace(/ /g, '_').toLowerCase())) {
+          throw new BadRequestException('Group name already exists');
+        }
+
+        const groupToFind = await this.groupPermissionsRepository.findOne({
+          where: {
+            organizationId: user.organizationId,
+            group: name,
+          },
+        });
+
+        if (groupToFind && groupToFind.id !== groupPermission.id) {
```

Comment:
if `groupToFind.id !== groupPermission.id` we can skip updation right?

Reply:
Yes